canton/community/ledger/ledger-common/src/main/scala/com/digitalasset/canton/ledger/error/PackageServiceErrors.scala

```diff
@@ -314,13 +314,35 @@ object PackageServiceErrors extends PackageServiceErrorGroup {
       )(implicit
           val loggingContext: ContextualizedErrorLogger
       ) extends DamlError(
-            cause = "A DAR with the same version number has previously been uploaded.",
-            extraContext = Map(
-              "uploadedPackage" -> uploadedPackage,
-              "existingPackage" -> existingPackage,
-              "packageVersion" -> packageVersion.toString,
-            ),
-          )
+          cause = "A DAR with the same version number has previously been uploaded.",
+          extraContext = Map(
+            "uploadedPackage" -> uploadedPackage,
+            "existingPackage" -> existingPackage,
+            "packageVersion" -> packageVersion.toString,
+          ),
+        )
+    }
+
+    @Explanation(
+      """This error indicates that the dry run flag was set in UploadDarRequest, so the uploaded Dar must not be uploaded."""
+    )
+    @Resolution("Upload with `dry_run: false` in the UploadDarRequest")
+    object DryRun
+        extends ErrorCode(
+          id = "DAR_DONT_UPLOAD_DUE_TO_DRY_RUN",
+          ErrorCategory.InvalidIndependentOfSystemState,
```

Comment:
Hm, it actually is a little bit dependent on state, as if theres a dar uploaded thats causes the upload check to fail, this error won't be thrown, another will. Its very minor and probably negligible, but I'm not sure how important these categories are

Reply:
I think it's invalid independent of system state in that if you ever see this error, it's because of something independent of the system state (the dryRun upload parameter).

If you don't see the error because another stateful check triggered a different error, then that error is stateful, not this one. If being pre-empted by a different stateful error made my own error stateful, then all errors would have to be classfied as stateful.

Reply:
Yeah okay I'm happy with that justification